commander.ts

```diff
@@ -15,6 +16,34 @@ import { Cmd, CmdVisitor, Exit, Print, Expression } from './ast/cmd';
 
 const tracer = getTracer('main');
 
+function mergeWarnings(
+  ...warnings: (ParseWarning | null)[]
+): ParseWarning | null {
+  let warning: ParseWarning | null = null;
+  for (const warn of warnings) {
+    if (warn) {
+      if (!warning) {
+        warning = new ParseWarning(warn.warnings);
+        continue;
+      }
+
+      warning.warnings = warning.warnings.concat(warn.warnings);
+    }
+  }
+
+  warning.warnings.sort((a, b) => {
```

Comment:
Warnings/errors are already sorted, so I should be able to implement a merge algorithm instead of doing a full blast sort. In TypeScript that might be slower than the native sort, but I'd still feel better about it.